sycl/include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp

```diff
@@ -289,14 +499,25 @@ struct joint_matrix_mad_impl<
 
 namespace experimental::matrix {
 
-template <typename Group, typename T, matrix_use MT, size_t NumRows,
+template <typename Group, typename T, matrix_use Use, size_t NumRows,
           size_t NumCols, matrix_layout Layout, access::address_space Space>
 void joint_matrix_load(
-    Group sg, joint_matrix<T, MT, NumRows, NumCols, Layout, Group> &res,
+    Group sg, joint_matrix<T, Use, NumRows, NumCols, Layout, Group> &res,
     multi_ptr<T, Space> src, size_t stride) {
-  sycl::ext::oneapi::detail::joint_matrix_load_impl<T, MT, NumRows, NumCols,
+#ifdef __SYCL_DEVICE_ONLY__
+#ifdef __NVPTX__
+  sycl::ext::oneapi::detail::joint_matrix_load_impl<T, Use, NumRows, NumCols,
                                                     Layout, Space>{}
       .load(res, src, stride);
+#endif
+#else
+  (void)sg;
+  (void)res;
+  (void)src;
+  (void)stride;
+  throw runtime_error("joint_matrix_load is not supported on host device.",
+                      PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__*/
```

Comment:
```suggestion
#if defined(__SYCL_DEVICE_ONLY__) && defined(__NVPTX__)
  sycl::ext::oneapi::detail::joint_matrix_load_impl<T, Use, NumRows, NumCols,
                                                    Layout, Space>{}
      .load(res, src, stride);
#else
  (void)sg;
  (void)res;
  (void)src;
  (void)stride;
  throw runtime_error("joint_matrix_load for tensor cores is only supported by CUDA devices.",
                      PI_INVALID_DEVICE);
#endif // defined(__SYCL_DEVICE_ONLY__) && defined(__NVPTX__)
```
I am not sure what the best error message would be but in the current version you could have warnings if `__SYCL_DEVICE_ONLY__` is defined and `__NVPTX__` isn't.

Note: Same applies to other similar functions.

Reply:
I made the suggested change.  It will be ideal to split the error into two cases: when __SYCL__DEVICE_ONLY__ is defined but a supported matrix extension backend isn't, and when HOST is used.  However it is probably a little easier working this out when the Intel and Nvidia matrix extension implementations are merged.